### PR TITLE
Bug 1209050 - Only expose languages with voices when screen reader is…

### PR DESCRIPTION
**Pull request by eeejay**
… enabled.

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=468e109e40501b381fbc80c47576496da50c3361

File: apps/system/js/accessibility.js [hunk 1]
```diff
@@ -229,6 +238,19 @@
       }
     },
 
+    setToSupportedLanguage: function ar_setToSupportedLanguage() {
```
Comment (yzen) [inline apps/system/js/accessibility.js]: A jsdoc comment would be nice.

File: shared/js/language_list.js [hunk 2]
```diff
@@ -111,13 +121,18 @@ exports.LanguageList = {
...
-      this._extendAdditional(langs, this._parseVersion(ver), addl);
-      return [langs, current];
-    }.bind(this));
+    ]).then(
+      function([langsFromFile, ver, current, qpsEnabled, srEnabled, addl]) {
+        var langs = this._copyObj(langsFromFile);
+        this._extendPseudo(langs, current, qpsEnabled);
+        this._extendAdditional(langs, this._parseVersion(ver), addl);
+        if (srEnabled) {
+          this._removeWithNoSpeech(langs);
```
Comment (yzen) [inline shared/js/language_list.js]: Just looking at the other functions here, could you pass srEnabled as an arg and check if it's enabled internally?

Comment (yzen) [inline shared/js/language_list.js]: ^ Nit

**Comment by yzen** [comment]
Looks like the tests are failing, I would also add tests for new functionality.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bf7ec0fe2656f25318404f343b36946a1a48f581

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cff73d8fa4baa396474db7c2a1a87fc6045bb5a0

File: apps/ftu/js/language.js [hunk 3]
```diff
@@ -8,13 +8,34 @@ var LanguageManager = {
   init: function init() {
     this.buildLanguageList();
     document.getElementById('languages').addEventListener('change', this);
-    this.settings.addObserver('language.current',
-      function updateDefaultLayouts(event) {
-        // the 2nd parameter is to reset the current enabled layouts
-        KeyboardHelper.changeDefaultLayouts(event.settingValue, true);
-      });
     window.addEventListener('localized',
```
Comment (sfoster) [inline apps/ftu/js/language.js]: Nit: if we're going to uninit, lets remove the localized and change event listeners in uninit too

File: apps/ftu/test/unit/language_test.js [hunk 4]
```diff
@@ -108,4 +112,31 @@ suite('languages >', function() {
...
+      LanguageManager.init();
+    });
+
+    suiteTeardown(function() {
+      LanguageManager.uninit();
+    });
+
+    test('enable screen reader', function(done) {
+      function afterEnabledHandler() {
+        window.removeEventListener("languagelistready", afterEnabledHandler);
```
Comment (sfoster) [inline apps/ftu/test/unit/language_test.js]: Nit: use single-quotes, this fails jshint.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6f13058f468f03d4da74c229438904474c3f7b46

**Pull request merged**